themes/default/content/blog/advanced-aws-networking-part-2/index.md

```diff
@@ -0,0 +1,534 @@
+
+First, add a file `spoke.py` to your Pulumi program. In Python, it's good practice to put each component resource along with its associated arguments class in a separate file from your main program as it makes for intuitive code organization:
+
+```python
+@dataclass
+class SpokeVpcArgs:
+    vpc_cidr_block: str
+    tgw_id: pulumi.Input[str]
+    tgw_route_table_id: pulumi.Input[str]
+
+
+class SpokeVpc(pulumi.ComponentResource):
+    def __init__(self, name: str, args: SpokeVpcArgs, opts: pulumi.ResourceOptions = None) -> None:
+        super().__init__("awsAdvancedNetworking:index:SpokeVpc", name, None, opts)
+```
+
+Then, create a VPC with 2 sets of subnets, both isolated (that is, with no pathway to the internet). One set of subnets will hold our spoke VPC workloads, and the other will hold our Transit Gateway attachment. We enable DNS hostnames and support so that we can later use VPC endpoints to enable [AWS Systems Manager](https://docs.aws.amazon.com/systems-manager/index.html) access to verify that centralized egress is working as expected without requiring direct host access via SSH:
+
+```python
+self.vpc = awsx.ec2.Vpc(
+    f"{self._name}-vpc",
+    awsx.ec2.VpcArgs(
+        cidr_block=args.vpc_cidr_block,
+        subnet_specs=[
+            awsx.ec2.SubnetSpecArgs(
+                name="private",
+                cidr_mask=28,
+                type=awsx.ec2.SubnetType.ISOLATED,
+            ),
+            awsx.ec2.SubnetSpecArgs(
+                name="tgw",
+                cidr_mask=28,
+                type=awsx.ec2.SubnetType.ISOLATED,
+            ),
+        ],
+        nat_gateways=awsx.ec2.NatGatewayConfigurationArgs(
+            strategy=awsx.ec2.NatGatewayStrategy.NONE,
+        ),
+        enable_dns_hostnames=True,
+        enable_dns_support=True,
+    )
+)
+```
+
+Then, attach the Transit Gateway to the appropriate subnets:
+
+```python
+self.tgw_attachment = aws.ec2transitgateway.VpcAttachment(
+    f"{name}-tgw-vpc-attachment",
+    aws.ec2transitgateway.VpcAttachmentArgs(
+        transit_gateway_id=args.tgw_id,
+        subnet_ids=tgw_subnets.apply(lambda x: x.ids),
+        vpc_id=self.vpc.vpc_id,
+        transit_gateway_default_route_table_association=False,
+        transit_gateway_default_route_table_propagation=False,
+        tags={
+            "Name": f"{name}",
+        },
+    ),
+    pulumi.ResourceOptions(
+        delete_before_replace=True,
+        depends_on=[self.vpc],
+        parent=self,
+    )
+)
+```
+
+Add [VPC endpoints](https://docs.aws.amazon.com/vpc/latest/privatelink/create-interface-endpoint.html) to allow Systems Manager access so that you can open an SSH connection from the AWS console without having to allow network access to the hosts. Because Pulumi uses real programming languages, you can use a plain old Python `for` loop to create these resources with less lines of code using familiar syntax:
```

Comment:
```suggestion
Add [VPC endpoints](https://docs.aws.amazon.com/vpc/latest/privatelink/create-interface-endpoint.html) to allow Systems Manager access to open an SSH connection from the AWS console without having to allow network access to the hosts. Because Pulumi uses real programming languages, you can use a Python `for` loop to create these resources with fewer lines of code using familiar syntax:
```